src/Lean/Server/FileWorker/RequestHandling.lean

```diff
@@ -92,16 +95,20 @@ partial def handleDefinition (goToType? : Bool) (p : TextDocumentPositionParams)
         if let some (ci, i) := t.hoverableInfoAt? hoverPos then
           if let Info.ofTermInfo ti := i then
             let mut expr := ti.expr
-            if goToType? then
+            if kind = type then
               expr ← ci.runMetaM i.lctx do
                 Meta.instantiateMVars (← Meta.inferType expr)
             if let some n := expr.constName? then
               return ← ci.runMetaM i.lctx <| locationLinksFromDecl i n
+          if let Info.ofFieldInfo fi := i then
+            if kind ≠ type then
```

Comment:
Should we not support going to the type of a structure field?

Reply:
I didn't add this since projections have Pi-types, but you're right, it is more convenient to just go to the codomain.